docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -580,59 +776,115 @@ public int Shape
 
     ![Stacks](media/stacks.png)
 
-<a name="textvisual"></a>
-## Text Visual
 
-The text visual renders text within a control.
 
-**Figure: Text visual**
+**Usage:**
+
+The following example illustrates how to draw a BevelledCube.
+   ```csharp
+   PrimitiveVisual _primitiveVisual = new PrimitiveVisual();
+   _primitiveVisual.Shape = PrimitiveVisualShapeType.BevelledCube;
+   _primitiveVisual.MixColor = new Vector4(0.4f, 0.4f, 1.0f, 1.0f);
+   // Two initial coord. indicate the top-left corner; the third one indicates the shift toward the observer
+   _primitiveVisual.LightPosition = new Vector3(0, 0, 1000);
+   _primitiveVisual.ScaleDimensions = new Vector3(1.0f, 0.5f, 1.4f);
+   _primitiveVisual.BevelPercentage = 0.5f;
+   _primitiveVisual.BevelSmoothness = 0.0f;
+
+   _visualView.AddVisual("BevelledCube", _primitiveVisual);
+
+   ```
+
+
+**Figure: PrimitiveVisual from the above code**
+
+![Primitive visual](media/visuals/PrimitiveVisualExample_BevelledCube.png)
+
+
+
+<!---------------------------------------------------------------------------------------------------------------------------->
+<a name="textVisual"></a>
+### TextVisual
 
-![Text visual](media/HelloWorld.png)
+The TextVisual renders text within a control.
 
-The following table lists the supported properties. The visual type is `Text` and the visual map is `TextVisual`.
 
-**Table: Text visual properties**
+The following table lists the supported properties:
+
+**Table: TextVisual properties**
+
+| Property              | Type                  | Required | Description                                                          |
+|-----------------------|-----------------------|----------|----------------------------------------------------------------------|
+| `Text`                | `string`              | Yes      | The text to display in UTF-8 encoding.                               |
+| `FontFamily`          | `string`              | No       | The requested font family to use.                                    |
+| `FontStyle`           | `PropertyMap`         | No       | The requested font style to use.                                     |
+| `PointSize`           | `float`               | Yes      | The font size in points.                                             |
+| `MultiLine`           | `bool`                | No       | Specifies whether to use a multi-line layout (default - `false`).    |
+| `HorizontalAlignment` | `HorizontalAlignment` | No       | The line horizontal alignment: `Begin` (default), `Center` or `End`. |
+| `VerticalAlignment`   | `VerticalAlignment`   | No       | The line vertical alignment: `Top` (default), `Center` or `Bottom`.  |
+| `TextColor`           | `Color`               | No       | The text color.                                                      |
+| `EnableMarkup`        | `bool`                | No       | Specifies whether markup processing is enabled (default is `false`). |
```

Comment:
Global: Please be consistent with styling in all tables ( default - false) or (default false): You can choose one styling and implement it in all tables.